interp/notation.ml

```diff
@@ -1979,9 +1979,19 @@ let make_notation_key from symbols =
 
 let decompose_notation_pure_key s =
   let len = String.length s in
+  let rec find_string_end n =
+    let next =
+      try String.index_from s (n+1) '"'
+      with Not_found -> assert false
```

Comment:
Test suite triggers this assertion on
```
(* Test string literals themselves with double quotes *)
Notation """t""""t""" := tt.
Check "t""t".
```
according to CI

Reply:
This error with `"t""t"` is caused by a typo three lines below (`find_string_end (n+2)` should be `find_string_end (next+2)`, now fixed).

However the problem in `allBytes.v` for the only printing notation `Notation "'""'" := ...` is different and a decision has to be taken.

I assumed that there were no need for a notation with a single `"`, because it cannot be parsed, but an only printing notation with a single double quotes seem legitimate.

The question is what encoding to use for string literals containing spaces in a notation, if we also want to support isolated double quotes. For instance, what distinct encoding to give to the list of two tokens `["\"";"\""]` (in Ocaml/C syntax) and to the single token `["\" \""]`?

Would the convention that for a string literal to contain spaces we expect it to be separated by spaces (as in `"x ""two words"" y"`) and that if want double quotes not part of string literal, we enclose it in single quotes, as in `"x 'foo"' y"`, but then excluding the possibililty of spaces (if we really want spaces in only printing, we can use a format)? Otherwise said, for the example above, one would have respectively `"'""' '""'` and `""" """` (even if both are awful to read, but I guess we cannot avoid such encoding to be easily parsable for a human).

Reply:
> `"'""' '""'`

Do you mean  `"'""' '""'"`?

Anyway, yes, this seems fine.  It is backwards compatible, and any way you might want the notation to print can be expressed with the help of format, right?

Reply:
Proposed changes done. New tests added. Documentation added. Now testing.